### PR TITLE
Remove usage of deprecated load methods

### PR DESCRIPTION
**Pull request by kian-thompson**
`ContainerRuntime.load(...)` and `FluidDataStoreRuntime.load(...)` have been deprecated for some time. Getting off these methods will help with the effort to move to the `entryPoint` pattern.

### CODE REVIEW

File: packages/framework/attributor/src/mixinAttributor.ts [hunk 1]
```diff
@@ -112,8 +112,46 @@ export const mixinAttributor = (Base: typeof ContainerRuntime = ContainerRuntime
 			runtimeOptions: IContainerRuntimeOptions | undefined = {},
 			containerScope: FluidObject | undefined = context.scope,
 			existing?: boolean | undefined,
-			ctor: typeof ContainerRuntime = ContainerRuntimeWithAttributor as unknown as typeof ContainerRuntime,
+			containerRuntimeCtor: typeof ContainerRuntime = ContainerRuntimeWithAttributor as unknown as typeof ContainerRuntime,
 		): Promise<ContainerRuntime> {
```
Comment (markfields) [inline packages/framework/attributor/src/mixinAttributor.ts:116]: Should we deprecate this `public static async load` function as well?  Feels weird to keep it around when it won't work (due to `provideEntryPoint` throwing)

File: packages/framework/attributor/src/test/mixinAttributor.spec.ts [hunk 2]
```diff
@@ -88,13 +88,15 @@ describe("mixinAttributor", () => {
...
 			context,
-			[],
-			undefined, // requestHandler
-			{}, // runtimeOptions
-			getScope(),
-		);
+			registryEntries: [],
+			containerScope: getScope(),
+			provideEntryPoint: async () => ({
+				myProp: "myValue",
```
Comment (markfields) [inline packages/framework/attributor/src/test/mixinAttributor.spec.ts:96]: Where did this come from?  Why not return `{}`?

File: packages/test/test-utils/src/testContainerRuntimeFactory.ts [hunk 3]
```diff
@@ -61,22 +61,25 @@ export const createTestContainerRuntimeFactory = (
...
-				buildRuntimeRequestHandler(
+				requestHandler: buildRuntimeRequestHandler(
 					// eslint-disable-next-line import/no-deprecated
 					defaultRouteRequestHandler("default"),
 					...this.requestHandlers,
 				),
-				this.runtimeOptions,
-				context.scope,
+				provideEntryPoint: () => {
+					throw new Error("TODO: AB#4993");
```
Comment (markfields) [inline packages/test/test-utils/src/testContainerRuntimeFactory.ts:77]: Why can't we properly implement this now?  Similar to what `defaultRouteRequestHandler` does?

File: examples/utils/example-utils/src/modelLoader/modelContainerRuntimeFactory.ts [hunk 4]
```diff
@@ -37,15 +37,17 @@ export abstract class ModelContainerRuntimeFactory<ModelType> implements IRuntim
...
 			context,
-			this.registryEntries,
+			registryEntries: this.registryEntries,
 			// eslint-disable-next-line import/no-deprecated
-			makeModelRequestHandler(this.createModel.bind(this)),
-			this.runtimeOptions,
-			undefined, // scope
+			requestHandler: makeModelRequestHandler(this.createModel.bind(this)),
+			provideEntryPoint: () => {
+				throw new Error("TODO: AB#4990");
```
Comment (markfields) [inline examples/utils/example-utils/src/modelLoader/modelContainerRuntimeFactory.ts:46]: If we can't address this in this PR, let's reword this error message.  AB# can go in a comment.  Same elsewhere

**Review by markfields — APPROVED** [review]
It would be awesome if we can actually implement `provideEntryPoint` in the cases where you're currently doing `throw new Error("TODO: AB#4990");`.  But if it's a fast-follow this sprint that's ok too.

File: packages/framework/attributor/src/mixinAttributor.ts (hunk 1 above)
Comment (kian-thompson) [reply packages/framework/attributor/src/mixinAttributor.ts:116]: This isn't technically exported/visible. This mixin pattern on static methods is very weird to me

File: packages/test/test-utils/src/testContainerRuntimeFactory.ts (hunk 3 above)
Comment (kian-thompson) [reply packages/test/test-utils/src/testContainerRuntimeFactory.ts:77]: I'd rather have this explicitly handled by the task to remove request pattern in the tests. Throwing an error can quickly let someone know entryPoint isn't yet handled by e2e tests so they don't need to do a ton of debugging.

**Pull request merged**